crates/sui-config/src/dynamic_transaction_signing_checks.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{
+    de::{self, Deserializer},
+    ser::Serializer,
+    Deserialize,
+};
+use serde_json::Value as JsonValue;
+use starlark::{
+    environment::{Globals, Module},
+    eval::Evaluator,
+    syntax::{AstModule, Dialect, DialectTypes},
+    values::{dict::AllocDict, AllocValue, Heap, Value},
+    ErrorKind,
+};
+use sui_types::{
+    base_types::ObjectRef,
+    signature::GenericSignature,
+    transaction::{InputObjectKind, TransactionData},
+};
+use tracing::warn;
+
+/// The name of the input variables that the transaction will have in the Starlark program.
+const TX_DATA_NAME: &str = "tx_data";
+const TX_SIGNERS_NAME: &str = "tx_signers";
+const TX_INPUT_OBJECTS_NAME: &str = "tx_input_objects";
+const TX_RECEIVING_OBJECTS_NAME: &str = "tx_receiving_objects";
+
+/// The dummy name of the Starlark file being executed. We will just be passing the string of the
+/// program in directly so this is not important but may appear in error messages.
+const STAR_INPUT_FILE_NAME: &str = "dummy.star";
```

Comment:
nit: maybe call it something that indicates it's for transaction signing checks, to make those potential error messages less confusing? Maybe you can use the name of the configuration it is set in or something.